.github/workflows/ci.yml

```diff
@@ -0,0 +1,120 @@
+---
+name: OpenWISP Monitoring CI Build
+
+on:
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches:
+      - master
+      - dev
+
+jobs:
+  build:
+    name: Python==${{ matrix.python-version }} | ${{ matrix.django-version }}
+    runs-on: ubuntu-20.04
+
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+      influxdb:
+        image: influxdb
+        ports:
+          - 8086:8086
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.6
+          - 3.7
+        django-version:
+          - django~=2.2
+          - django~=3.0
+          - django~=3.1
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install system packages
+        run: |
+          sudo apt-get update && sudo apt-get upgrade
+          sudo apt-get -qq install \
+          sqlite3 \
+          fping \
+          gdal-bin \
+          libproj-dev \
+          libgeos-dev \
+          libspatialite-dev \
+          spatialite-bin \
+          libsqlite3-mod-spatialite
+
+      - name: Install python dependencies
+        run: |
+          pip install -U "pip==20.2.4" wheel setuptools
+          pip install ${{ matrix.django-version }}
+          pip install -U -r requirements-test.txt
+
+      - name: Install openwisp-monitoring
+        run: pip install -e .
+
+      - name: QA checks
+        run: |
+          ./run-qa-checks
+
+      - name: Tests
+        run: |
+          SAMPLE_APP=1 coverage run --source=openwisp_monitoring runtests.py
+          coverage run -a --source=openwisp_monitoring runtests.py
+
+      - name: Upload Coverage
+        run: coveralls --service=github
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_FLAG_NAME: |
+           python-${{ matrix.python-version }}-${{ matrix.django-version }}
+          COVERALLS_PARALLEL: true
+
+  deploy:
+    name: Deploy Docker Image
+    runs-on: ubuntu-20.04
```

Comment:
@pandafy Should I run this only after build is completed??

Reply:
Yes @devkapilbansal. It will prevent pushing broken images to DockerHub.